omnichain/swap/contracts/Swap.sol

```diff
@@ -47,26 +49,30 @@ contract Swap is zContract {
             recipientAddress = recipient;
         }
 
-        uint256 outputAmount = SwapHelperLib._doSwap(
+        (address gasZRC20, uint256 gasFee) = IZRC20(targetTokenAddress)
+            .withdrawGasFee();
+
+        inputForGas = SwapHelperLib.swapTokensForExactTokens(
+            systemContract.wZetaContractAddress(),
+            systemContract.uniswapv2FactoryAddress(),
+            systemContract.uniswapv2Router02Address(),
+            zrc20,
+            gasFee,
+            gasZRC20,
+            amount
+        );
+
+        outputAmount = SwapHelperLib._doSwap(
             systemContract.wZetaContractAddress(),
             systemContract.uniswapv2FactoryAddress(),
             systemContract.uniswapv2Router02Address(),
             zrc20,
-            amount,
+            amount - inputForGas,
```

Comment:
this looks wrong... 
amount: is expressed in inbound token
inputForGas: is expressed in destination gas
outputAmount: is expressed in destination token

then amount - inputForGas are mixing tokens and makes no sense. You should calculate some how (current balance?) how much of amount was used to do the swap and that number use in line 70

Reply:
`inputForGas` is in inbound token as well:

https://github.com/zeta-chain/toolkit/blob/477f9d9f5a5a69249102847f694f78410ef89476/contracts/SwapHelperLib.sol#L163

Reply:
`swapTokensForExactTokens` returns `amounts[0]` which is as far as I understand the token your putting in, which is inbound token in our case.